xilem/examples/external_event_loop.rs

```diff
@@ -0,0 +1,147 @@
+// Copyright 2024 the Xilem Authors
+// SPDX-License-Identifier: Apache-2.0
+
+use masonry::{
+    app_driver::AppDriver,
+    widget::{CrossAxisAlignment, MainAxisAlignment},
+    ArcStr,
+};
+use winit::{
+    application::ApplicationHandler,
+    error::EventLoopError,
+    event::ElementState,
+    keyboard::{KeyCode, PhysicalKey},
+};
+use xilem::{
+    view::{button, flex, label, sized_box},
+    EventLoop, WidgetView, Xilem,
+};
+
+/// A component to make a bigger than usual button
+fn big_button(
+    label: impl Into<ArcStr>,
+    callback: impl Fn(&mut i32) + Send + Sync + 'static,
+) -> impl WidgetView<i32> {
+    sized_box(button(label, callback)).width(40.).height(40.)
+}
+
+fn app_logic(data: &mut i32) -> impl WidgetView<i32> {
+    flex((
+        big_button("-", |data| {
+            *data -= 1;
+        }),
+        label(format!("count: {}", data)).text_size(32.),
+        big_button("+", |data| {
+            *data += 1;
+        }),
+    ))
+    .direction(xilem::Axis::Horizontal)
+    .cross_axis_alignment(CrossAxisAlignment::Center)
+    .main_axis_alignment(MainAxisAlignment::Center)
+}
+
+struct AppInterface {
```

Comment:
`AppInterface` is not a very evocative name here. I think `App` might be better, but still not great. Maybe we could use `ExistingApp` :P

Reply:
Makes sense. Will do.